app/src/main/java/org/wikipedia/dataclient/restbase/RbServiceError.kt

```diff
@@ -0,0 +1,28 @@
+package org.wikipedia.dataclient.restbase
+
+import org.apache.commons.lang3.StringUtils
+import org.wikipedia.dataclient.ServiceError
+import org.wikipedia.json.GsonUnmarshaller
+
+class RbServiceError : ServiceError {
+
+    private val type: String? = null
+    private val title: String = ""
+    private val detail: String? = null
+    private val method: String? = null
+    private val uri: String? = null
+
+    override fun getTitle(): String {
+        return StringUtils.defaultString(title)
+    }
+
+    override fun getDetails(): String {
+        return StringUtils.defaultString(detail)
+    }
+
+    companion object {
+        fun create(rspBody: String): RbServiceError {
+            return GsonUnmarshaller.unmarshal(RbServiceError::class.java, rspBody)
+        }
+    }
```

Comment:
Would it be possible to convert `ServiceError` to Kotlin as well, and the `getTitle()` can be combined with the `val title` variable.

Reply:
Done @cooltey